.github/workflows/pull-request.yaml

```diff
@@ -6,6 +6,35 @@ on:
       - main
 
 jobs:
+  version-check:
+    name: Version Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Check All Versions Match
+        run: |
+          pkg_version=$(grep -P -o 'version = "\d+\.\d+\.\d+"' pyproject.toml | grep -P -o '\d+\.\d+\.\d+')
+          rock_version=$(grep -P -o 'version: "\d+\.\d+\.\d+"' rockcraft.yaml | grep -P -o '\d+\.\d+\.\d+')
+          rock_pkg_version=$(grep -P -o 'cos-alerter==\d+\.\d+\.\d+' rockcraft.yaml | grep -P -o '\d+\.\d+\.\d+')
+          [ "${pkg_version}" == "${rock_version}" ]
+          [ "${rock_version}" == "${rock_pkg_version}" ]
+  changelog-check:
+    name: Changelog Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
0 is full depth. You need 1. Which is the default iirc.
```suggestion
```

Reply:
I think we need full depth so we get all the tags. We check against the tags to see if this is a new version.